lte/gateway/c/core/oai/tasks/async_grpc_service/grpc_async_service_task.cpp

```diff
@@ -0,0 +1,232 @@
+/**
```

Comment:
@uri200 @rsarwad a lot of code and class definitions look like copied from sessiond. why don't we put these into a common folder across c/c++ services?

Reply:
Hey @ulaskozat  we can do that, but I think it would be better to try to do this on a different PR

The issue this PR fixes has been blocking some s6a regression tests since inception of the service. So it would be nice to have Cancel Location test pass eventually

Reply:
Yes @ulaskozat , we can create a separate PR